zebra/if_netlink.c

```diff
@@ -1721,18 +1725,27 @@ int netlink_link_change(struct nlmsghdr *h, ns_id_t ns_id, int startup)
 
 					/* Update EVPN VNI when SVI MAC change
 					 */
-					if (IS_ZEBRA_IF_VLAN(ifp) &&
-					    memcmp(old_hw_addr, ifp->hw_addr,
-						   INTERFACE_HWADDR_MAX)) {
+					if (memcmp(old_hw_addr, ifp->hw_addr,
+						   INTERFACE_HWADDR_MAX))
+						mac_updated = true;
+					if (mac_updated
+					    && IS_ZEBRA_IF_VLAN(ifp)) {
 						struct interface *link_if;
 
-						link_if =
-						if_lookup_by_index_per_ns(
-						zebra_ns_lookup(NS_DEFAULT),
-								link_ifindex);
+						link_if = if_lookup_by_index_per_ns(
+							zebra_ns_lookup(
+								NS_DEFAULT),
+							link_ifindex);
 						if (link_if)
-							zebra_vxlan_svi_up(ifp,
-								link_if);
+							zebra_vxlan_svi_up(
+								ifp, link_if);
+					} else if (mac_updated
+						   && IS_ZEBRA_IF_BRIDGE(ifp)) {
+						zlog_debug(
+							"Intf %s(%u) bridge changed MAC address",
+							name, ifp->ifindex);
+						chgflags =
+							ZEBRA_BRIDGE_MASTER_MAC_CHANGE;
```

Comment:
same line ?

Reply:
actually, polycheata will complain. so, not possible.